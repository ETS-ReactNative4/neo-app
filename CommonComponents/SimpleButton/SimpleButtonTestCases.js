import React from "react";
import SimpleButton from "./SimpleButton";
import constants from "../../constants/constants";

const SimpleButtonTestCases = [
  {
    title: "Simple button renders correctly",
    Component: <SimpleButton text={"Click me"} textColor={"white"} />
  },
  {
    title: "Simple button with click action",
    Component: (
      <SimpleButton
        text={"Click me"}
        textColor={"white"}
        action={() => alert("clicked")}
      />
    )
  },
  {
    title: "Simple button transparent background with border",
    Component: (
      <SimpleButton
        text={"Click me"}
        textColor={constants.firstColor}
        color={"transparent"}
        hasBorder={true}
      />
    )
  },
  {
    title: "Simple button button with left icon",
    Component: (
      <SimpleButton
        text={"Click me"}
        textColor={"white"}
        icon={constants.busIcon}
        iconSize={14}
      />
    )
  },
  {
    title: "Simple button button with right icon",
    Component: (
      <SimpleButton
        text={"Click me"}
        textColor={"white"}
        icon={constants.busIcon}
        iconSize={14}
        rightIcon={true}
      />
    )
  },
  {
    title: "Simple button with underlay color",
    Component: (
      <SimpleButton
        text={"Click me"}
        textColor={"white"}
        underlayColor={constants.secondColor}
      />
    )
  }
];

export default SimpleButtonTestCases;
