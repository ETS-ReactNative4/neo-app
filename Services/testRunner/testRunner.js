import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
