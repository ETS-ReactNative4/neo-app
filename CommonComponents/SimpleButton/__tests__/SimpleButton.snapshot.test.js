import React from "react";

import SimpleButtonTestCases from "../SimpleButtonTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

SimpleButtonTestCases.map(testRunner);
