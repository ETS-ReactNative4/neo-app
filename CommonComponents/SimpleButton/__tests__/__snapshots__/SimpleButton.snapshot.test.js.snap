// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple button button with left icon 1`] = `
<Unknown
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,198,132,1)",
        "borderRadius": 4,
        "height": 40,
        "width": 160,
      },
      Object {},
    ]
  }
  underlayColor="white"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {},
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 14,
          },
          undefined,
          Object {
            "fontFamily": "voyager",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 17,
            "lineHeight": 17,
            "marginTop": 2,
          },
          Object {
            "color": "white",
          },
          Object {
            "marginLeft": 8,
          },
        ]
      }
    >
      Click me
    </Text>
  </View>
</Unknown>
`;

exports[`Simple button button with right icon 1`] = `
<Unknown
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,198,132,1)",
        "borderRadius": 4,
        "height": 40,
        "width": 160,
      },
      Object {},
    ]
  }
  underlayColor="white"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "flexDirection": "row-reverse",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 14,
          },
          undefined,
          Object {
            "fontFamily": "voyager",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 17,
            "lineHeight": 17,
            "marginTop": 2,
          },
          Object {
            "color": "white",
          },
          Object {
            "marginLeft": 8,
          },
        ]
      }
    >
      Click me
    </Text>
  </View>
</Unknown>
`;

exports[`Simple button renders correctly 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,198,132,1)",
        "borderRadius": 4,
        "height": 40,
        "width": 160,
      },
      Object {},
    ]
  }
  underlayColor="white"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 17,
            "lineHeight": 17,
            "marginTop": 2,
          },
          Object {
            "color": "white",
          },
          Object {},
        ]
      }
    >
      Click me
    </Text>
  </View>
</TouchableHighlight>
`;

exports[`Simple button transparent background with border 1`] = `
<Unknown
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,198,132,1)",
        "borderRadius": 4,
        "height": 40,
        "width": 160,
      },
      Object {
        "backgroundColor": "transparent",
        "borderColor": "rgba(0,198,132,1)",
        "borderWidth": 1.2,
      },
    ]
  }
  underlayColor="white"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 17,
            "lineHeight": 17,
            "marginTop": 2,
          },
          Object {
            "color": "rgba(0,198,132,1)",
          },
          Object {},
        ]
      }
    >
      Click me
    </Text>
  </View>
</Unknown>
`;

exports[`Simple button with click action 1`] = `
<Unknown
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,198,132,1)",
        "borderRadius": 4,
        "height": 40,
        "width": 160,
      },
      Object {},
    ]
  }
  underlayColor="white"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 17,
            "lineHeight": 17,
            "marginTop": 2,
          },
          Object {
            "color": "white",
          },
          Object {},
        ]
      }
    >
      Click me
    </Text>
  </View>
</Unknown>
`;

exports[`Simple button with underlay color 1`] = `
<Unknown
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,198,132,1)",
        "borderRadius": 4,
        "height": 40,
        "width": 160,
      },
      Object {},
    ]
  }
  underlayColor="rgba(255, 201, 51, 1)"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 17,
            "lineHeight": 17,
            "marginTop": 2,
          },
          Object {
            "color": "white",
          },
          Object {},
        ]
      }
    >
      Click me
    </Text>
  </View>
</Unknown>
`;
